x-pack/plugins/siem/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -4,18 +4,21 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { ListClient } from '../../../../../lists/server';
 import { AlertServices } from '../../../../../alerting/server';
 import { RuleAlertAction } from '../../../../common/detection_engine/types';
-import { RuleTypeParams, RefreshTypes } from '../types';
+import { RuleTypeParams, RefreshTypes, RuleAlertParams } from '../types';
 import { Logger } from '../../../../../../../src/core/server';
 import { singleSearchAfter } from './single_search_after';
 import { singleBulkCreate } from './single_bulk_create';
 import { SignalSearchResponse } from './types';
+import { filterEventsAgainstList } from './filter_events_with_list';
 
 interface SearchAfterAndBulkCreateParams {
-  someResult: SignalSearchResponse;
   ruleParams: RuleTypeParams;
   services: AlertServices;
+  listClient: ListClient | undefined; // TODO: undefined is for temporary development, remove before merged
```

Comment:
Is it time to remove the `undefined` now?

Reply:
I have `undefined` for the case where lists is not a required plugin. I can change this if we are ready to make lists required for SIEM to work.

Reply:
I think that part is still under debate about how we handle this globally per some conversations around slack medium.

For now I think you should use your best judgment and if this is still optionally undefined because we are pushing it further down that stack as optional and the code doesn't look too gross it is ok for now.

Reply:
I have two if statements one [in the executor](https://github.com/elastic/kibana/pull/65372/files#diff-815ff3ab6e173c450113d571369263b5R212-R225) and another inside of the `searchAfterBulkCreate` [function](https://github.com/elastic/kibana/pull/65372/files#diff-9dd6171b96a38383ebb6b961e555ba96R141-R151). I'm thinking keeping these two statements wouldn't add too much "grossness" but I'm open to enforcing lists as required in order to register the `siem.signals` alert type by adding a null check inside of the siem plugin setup function.

Also, it's a good thing you pointed this out as I wouldn't have realized I hadn't removed [this line](https://github.com/elastic/kibana/pull/65372/files#diff-c2a9bc9178eed0b22770ab3eec4290feR174) when I was debugging something 😅.